packages/next/src/shared/lib/router/action-queue.ts

```diff
@@ -189,5 +191,22 @@ export function createMutableActionQueue(
     last: null,
   }
 
+  if (typeof window !== 'undefined') {
+    // The action queue is lazily created on hydration, but after that point
+    // it doesn't change. So we can store it in a global rather than pass
+    // it around everywhere via props/context.
+    if (globalActionQueue !== null) {
+      throw new Error(
```

Comment:
nit: we have an [`InvariantError`](https://github.com/vercel/next.js/blob/07e10789e12939bad34c46ae6dfb1cc5598511cd/packages/next/src/shared/lib/invariant-error.ts#L1) for these types of errors though I don't have a strong opinion on if we should use it

Reply:
Is it used for anything other than setting the name property to "InvariantError"? Does it change how the error gets reported?

Reply:
I believe it's purely to have a standardized error message for invariants